pymess/models/dialer.py

```diff
@@ -0,0 +1,121 @@
+    template = models.ForeignKey(settings.DIALER_TEMPLATE_MODEL, verbose_name=_('template'), blank=True, null=True,
+                                 on_delete=models.SET_NULL, related_name='dialer_messages')
+    state = models.IntegerField(verbose_name=_('state'), null=False, blank=False, choices=STATE.choices, editable=False)
+    backend = models.CharField(verbose_name=_('backend'), null=True, blank=True, editable=False, max_length=250)
+    error = models.TextField(verbose_name=_('error'), null=True, blank=True, editable=False)
+    extra_data = JSONField(verbose_name=_('extra data'), null=True, blank=True, editable=False)
+    extra_sender_data = JSONField(verbose_name=_('extra sender data'), null=True, blank=True, editable=False)
+    tag = models.SlugField(verbose_name=_('tag'), null=True, blank=True, editable=False)
+
+    def clean_recipient(self):
+        self.recipient = normalize_phone_number(force_text(self.recipient))
+
+    def clean_content(self):
+        pass
+
+    @property
+    def failed(self):
+        return self.state == self.STATE.ERROR
+
+    def __str__(self):
+        return self.recipient
+
+    class Meta:
+        abstract = True
+        verbose_name = _('dialer message')
+        verbose_name_plural = _('dialer messages')
+        ordering = ('-created_at',)
+
+
+class DialerMessage(AbstractDialerMessage):
+    pass
+
+
+class AbstractDialerTemplate(SmartModel):
+
+    slug = models.SlugField(verbose_name=_('slug'), max_length=100, null=False, blank=False, editable=False,
+                            primary_key=True)
+    body = models.TextField(verbose_name=_('message body'), null=True, blank=False)
+
+    def _update_context_data(self, context_data):
+        return context_data
+
+    def clean_body(self, context_data=None):
+        try:
+            self.render_body(context_data or {})
+        except (TemplateSyntaxError, TemplateDoesNotExist) as ex:
+            raise ValidationError(ugettext('Error during template body rendering: "{}"').format(ex))
+
+    def get_body(self):
+        return self.body
+
+    def render_body(self, context_data):
+        context_data = self._update_context_data(context_data)
+        return Template(self.get_body()).render(Context(context_data))
+
+    def can_send(self, recipient, context_data):
+        return True
+
+    def send(self, recipient, context_data, related_objects=None, tag=None):
+        if self.can_send(recipient, context_data):
+            return get_dialer_sender().send(
+                recipient,
+                self.render_body(context_data),
+                template=self,
+                related_objects=related_objects,
+                tag=tag
+            )
+        else:
+            return None
+
+    def __str__(self):
+        return self.slug
+
+    class Meta:
+        abstract = True
+        verbose_name = _('SMS template')
+        verbose_name_plural = _('SMS templates')
```

Comment:
SMS?

Reply:
nezapomeň pak vygenerovat překlady

Reply:
sory, toto mi nejako uniklo, lebo som to kontroloval asi 3x :\

Reply:
preklady som nejak vygeneroval, ale z pochopitelnych dovodov ich uprav ;)